tcmur_cmd_handler.c

```diff
@@ -144,6 +144,124 @@ static int write_work_fn(struct tcmu_device *dev,
 				block_size * tcmu_get_lba(cmd->cdb));
 }
 
+static void handle_write_same_16_null_cbk(struct tcmu_device *dev,
+					  struct tcmulib_cmd *cmd,
+					  int ret)
+{
+}
+
+static int write_same_16_work_fn(struct tcmu_device *dev,
+				 struct tcmulib_cmd *cmd)
+{
+	struct tcmur_handler *rhandler = tcmu_get_runner_handler(dev);
+	uint32_t block_size = tcmu_get_dev_block_size(dev);
+	uint32_t lba_cnt = tcmu_get_xfer_length(cmd->cdb);
+	uint64_t start_lba = tcmu_get_lba(cmd->cdb);
+	uint64_t end_lba;
+	uint32_t max_lbas, copy_lbas;
+	uint32_t max_sectors;
+	int ret;
+
+	end_lba = start_lba + lba_cnt;
+
+	max_sectors = tcmu_get_attribute(dev, "hw_max_sectors");
```

Comment:
Sorry for late.

As I know for bdevs, the hw_max_sectors usually will be decided by the limitation of the bdevs controller's DMA, which will allow a maximum sectors/blocks to be transfered each IO. This will be reduce the IO merge count or split count.

For the file, this will be similar and more or less decided by the filesystem's limitation.